spyder/api/plugins.py

```diff
@@ -154,6 +154,10 @@ class SpyderPlugin(BasePlugin):
     # Status: Optional
     CONFIGWIDGET_CLASS = None
 
+    # The plugin does not have a UI component
+    # Status: Optional
+    NO_WIDGET = True
```

Comment:
There's no need of this. You need to check if the plugin has a `dockwidget` attribute. Only SpyderPluginWidget's have one.